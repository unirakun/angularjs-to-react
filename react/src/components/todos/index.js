export { default } from './todos.container'
