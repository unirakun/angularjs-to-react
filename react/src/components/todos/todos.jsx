import React from 'react'
import PropTypes from 'prop-types'
import { onlyUpdateForPropTypes } from 'recompose'
import Todo from 'components/todo'

const Todos = ({ todos, allCompleted, onCompleteAll }) => {
  return (
    <section className="main">
      {todos.length > 0 &&
        <input
          className="toggle-all"
          type="checkbox"
          onChange={onCompleteAll}
          checked={allCompleted}
        />
      }
      <ul className="todo-list" >
        {todos.map(todo => <Todo key={todo} id={todo} />)}
      </ul>
    </section>
  )
}

Todos.propTypes = {
  todos: PropTypes.array,
  allCompleted: PropTypes.bool.isRequired,
  onCompleteAll: PropTypes.func.isRequired,
}

Todos.defaultProps = {
  todos: [],
}

export default onlyUpdateForPropTypes(Todos)
