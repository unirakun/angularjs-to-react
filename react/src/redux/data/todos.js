import { keyValue } from 'k-redux-factory'
import { mapPayload } from 'k-redux-factory/helpers'

/* Todos Reducer  with her mapper */
export const todosMapper = todo => ({ id: new Date().getTime(), todo: todo.trim() })
export default keyValue({ pre: [mapPayload(/@@krf\/ADD_TODOS/)(todosMapper)] })({ path: 'data', key: 'id', name: 'todos' })
