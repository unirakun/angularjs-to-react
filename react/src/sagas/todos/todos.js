import { call, put, select } from 'redux-saga/effects'
import store from 'redux/data/todos'
import completed from 'redux/ui/completed'
import editing from 'redux/ui/editing'

export function goToAngular() {
  window.location = '/angularjs'
}

export function* remove(todoId) {
  yield put(store.remove(todoId))
  yield put(editing.remove(todoId))
  yield put(completed.remove(todoId))
}

export function* update(todoId) {
  const todo = yield select(editing.get(todoId))
  if (!todo) return
  if (!todo.todo || !todo.todo.trim()) yield call(remove, todoId)
  else yield put(store.update(todo))
  yield put(editing.remove(todoId))
}

export function* complete(todo) {
  yield put(completed.add(todo))
}

export function* completeAll(check) {
  const todos = yield select(store.getAsArray)
  yield* todos.map(todo => call(complete, { ...todo, completed: check }))
}

export function* clearCompleted() {
  const completeTodos = yield select(completed.getBy('completed', true))
  const completedId = completeTodos.map(todo => todo.id)
  yield call(remove, completedId)
}
