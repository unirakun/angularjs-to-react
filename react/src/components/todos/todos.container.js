import { connect } from 'react-redux'
import todos from 'redux/data/todos'
import completed from 'redux/ui/completed'
import action from 'components/actionFactory'
import Component from './todos'

export const mapStateToProps = (state) => {
  return {
    todos: todos.getKeys(state),
    allCompleted: completed.getBy('completed', true)(state).length === todos.getLength(state),
  }
}

export const mapDispatchToProps = (dispatch) => {
  return {
    onCompleteAll: e => dispatch(action('COMPLETE_ALL_CHANGED', e.target.checked)),
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Component)
