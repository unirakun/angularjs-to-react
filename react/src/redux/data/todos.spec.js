/* eslint-env jest */
import { todosMapper } from './todos'

describe('redux/todos', () => {
  describe('todosMapper', () => {
    it('should create todo when todo is empty', () => {
      expect(todosMapper('')).toMatchSnapshot()
    })

    it('should create todo when todo is blank', () => {
      expect(todosMapper('  ')).toMatchSnapshot()
    })

    it('should create todo', () => {
      expect(todosMapper('todo')).toMatchSnapshot()
    })
  })
})
