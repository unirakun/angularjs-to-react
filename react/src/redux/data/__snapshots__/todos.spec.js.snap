// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redux/todos todosMapper should create todo 1`] = `
Object {
  "id": 1451606400000,
  "todo": "todo",
}
`;

exports[`redux/todos todosMapper should create todo when todo is blank 1`] = `
Object {
  "id": 1451606400000,
  "todo": "",
}
`;

exports[`redux/todos todosMapper should create todo when todo is empty 1`] = `
Object {
  "id": 1451606400000,
  "todo": "",
}
`;
